src/lib/pages/code-details/components/json-schema/SchemaPanel.tsx

```diff
@@ -0,0 +1,56 @@
+import { Flex, Text } from "@chakra-ui/react";
+
+import { AttachSchemaCard } from "lib/components/json-schema";
+import JsonReadOnly from "lib/components/json/JsonReadOnly";
+import type { Option } from "lib/types";
+import { jsonPrettify } from "lib/utils";
+
+interface SchemaPanelProps {
+  codeId: number;
+  codeHash: string;
+  attached: boolean;
+  schema: Option<object>;
+  openDrawer: () => void;
+}
+
+export const SchemaPanel = ({
+  codeId,
+  codeHash,
+  attached,
```

Comment:
I think it's no need to pass `attached` here. We can determine its value based on the schema.

Reply:
`attached` is needed, since some message schema can be `null`